### PR TITLE
Kb urls

### PR DESCRIPTION
**Pull request by mgerring**

### CODE REVIEW

File: server/models/Module.py [hunk 1]
```diff
@@ -28,13 +28,16 @@ class Meta:
...
+    link = models.URLField('link', max_length=200, default="")
 
     # icon name associated with module
     icon = models.CharField('icon', max_length=20, default='url')
 
     # Does this model bring in external data?
     loads_data = models.BooleanField('loads_data', default=False)
 
+    # URL for the module's documentation, defaults to our knowledge base root
+    help_url = models.URLField('help_url', max_length=200, default=False)
```
Comment (jstray) [inline server/models/Module.py]: What is default=False, and why False here and '' for link field?
Also, I suspect we want only partial (relative)  URLS for the help_url (see below), will URLField still handle that?

File: server/serializers.py [hunk 2]
```diff
@@ -32,9 +33,14 @@ class Meta:
 
 
 class ModuleSerializer(serializers.ModelSerializer):
+    help_url = serializers.SerializerMethodField()
+
+    def get_help_url(self, obj):
+        return obj.help_url if obj.help_url else KB_ROOT_URL
```
Comment (jstray) [inline server/serializers.py]: Don't we have to add KB_ROOT_URL to help_url? I am envisioning that the URL in the json will be relative, so just 'add-data/add-from-url'. The idea is that we can seamlessly rebase the KB links, if e.g. we move off Intercom

File: server/serializers.py [hunk 3]
```diff
@@ -36,7 +37,13 @@ class ModuleSerializer(serializers.ModelSerializer):
     help_url = serializers.SerializerMethodField()
 
     def get_help_url(self, obj):
-        return obj.help_url if obj.help_url else KB_ROOT_URL
+        url_pattern = re.compile('^http(?:s?)://', re.IGNORECASE)
```
Comment (jstray) [inline server/serializers.py]: btw you can just re.match() for a one-off, no need to compile

File: server/serializers.py [hunk 4]
```diff
@@ -36,7 +37,13 @@ class ModuleSerializer(serializers.ModelSerializer):
...
+
+        if re.search(url_pattern, obj.help_url):
+            return obj.help_url
+
+        return f"{KB_ROOT_URL}{obj.help_url}"
```
Comment (jstray) [inline server/serializers.py]: I had to look up 'f'. Why not just +?

Comment (mgerring) [reply server/serializers.py]: didn't realize you could do that, I haven't dug in to the full breadth of python 3 format strings

**Pull request merged**